### ISSUE TITLE
FEATURE: Set log_file in DB log if not writing file logs

### ISSUE THREAD

**Issue by RasmusSkytte** [ssi-dk/SCDB#64]
**Describe the feature**
`Logger` currently uses a dummy `log_filename` if it is not writing log files to disk. 
This means that the data base logs ends up with a "dummy" variable in the `log_file` column that does not have a external meaning.
Ideally, the value here should just be `NULL` (since there is no `log_file` associated to the DB log entry)
The requested feature is to ensure the `log_file` column of the DB logs matches the value of `Logger$log_filename`.
 

**What use will this feature have?**
Implementing this feature will reduce confusion surrounding the `log_file` column in the DB logs

**Who should add the feature?**

Asignee

**Additional context**

N/A

**Comment by marcusmunch** [comment]
It is true that `log_filename` does not make "human" sense, but we still use it for identifying the "current" row during `update_snapshot` operation.

After looking a bit into it, I would assume it's neatly done with a [`Logger$finalize()`](https://adv-r.hadley.nz/r6.html#finalizer) function. I did a quick test similar to the example in the link, and it looks like it works as advertised, removing the file while (in my test) RStudio was exiting both "expectedly" (closing the window) and "semi-crashing" the R session by calling `q()` within my script 😄 

In other words, I should be able to implement this pretty quickly.
